### ISSUE TITLE
Cox and KM give different survivals in 0.10.0

### ISSUE THREAD

**Issue by disimone** [CamDavidsonPilon/lifelines#294]
Hi,

sorry for reposting in a new issue, but I am not sure if #289 (now closed) is still being watched :)

Here are my steps:

1) prepare the data, scaling the continuous variables with sklearn's minmaxscaler. I also tried to remove my own scaling, but the results don't really change
2) pass the data through patsy and prepare the input to lifelines
3) fit a cox model
4) obtain the mean matrix (myPatsyMatrix.mean()) and use it to predict the Cox lifetime
4) perform a KM mesurement on the same population
6) compare the cox prediction and the KM measurement

With the previous version of lifelines, things were ok as long as I specified "normalize=False". Now the cox prediction and the KM measurement are significantly off.

I had a look at the changes in 0.10.0, and I am not sure I understand what is being done.

What I see in the code is:

1) you apply some transformation to the inputs (using `utils.normalize(fitData, fitDataMean, fitDataStd)`) and fit the model. this makes my own use of the minmaxscaler ineffective, but I don't see how it could create problems.
2) at prediction time, the user gives you a set of values for the features of the model. What should happen now is that you use the same transformation you used before fitting, e.g. `utils.normalize(newData, fitDataMean, fitDataStd)`, then feed the result to the calculation.

If I follow the execution flow starting e.g. from CoxPHFitter::predict_survival_function, I can't really see where you apply the transformation to the inputs X. Or am I misunderstanding the code?

Assuming I am missing something in the code and all transformations are done correctly, this means that, when predicting on a point that is exactly the mean of the fitted dataset, all partial hazards do not contribute anymore, and the only relevant term is the baseline. Doesn't this mean that the baseline survival from cox  should be fairly similar to KM's measurement?

Andrea.

**Comment by CamDavidsonPilon** [comment]
Looking into this, I think I did miss a second normalization procedure.

**Comment by CamDavidsonPilon** [comment]
Hypothesis: If you _do_ normalize you dataset prior, you should receive the expected results. Is that correct?

**Comment by disimone** [comment]
Hi,

to be completely honest, I think kyou should avoid all normalizations. Just leave that to the users, like sklearn does. It is much less error prone, and users who do not know how and why data should be normalized, should probably not be using your library anyway...

To answer your question: it depends on what normalization I and you apply. if we apply the same, and you are really missing one normalization, then I should just add an extra normalization to my inputs X to get the correct resutls. Unfortunately, I tried this, and it does not work.

I think the problem is at the fitting stage. As I said, with the normalization you chose to apply (mean 0 and std 1), the average lifeline of the whole training set must be identical to the baseline (since all the exponentials give factors of 1 to the survival function), which then must be farily similar to KM. This is not happening at the moment. The baseline from cox is significantly different from the KM measurement, which means that somehow you are not applying the normalization correctly in the fit stage (or between the fit and the calculation of the baseline survival curve).

Let me know if I can help you testing anything :)

Thanks for your help,

Andrea.
